### DISCUSSION TITLE
[일지] 6월 4일 (금) 오후 회의록

### DISCUSSION THREAD

**Discussion by olenmg** [bcaitech1/p4-opt-5-vibrhanium-#50] [일지]
### PR 점검
- `train_optuna.py(optuna_search.py)` 버그 fix (재경)
    - `train_optuna.py` -> `optuna_search.py`
    - `optuna_train.py`는 search한 space에 대한 학습을 담당 
    - `expand_ratio` 등을 optuna가 그동안 tracking하지 못했던 버그 수정
        - optuna가 tracking 하도록 수정 (X) / tracking 안해도 저장되도록 수정 (O)
        - -> **optuna가 tracking 하도록 수정하겠습니당**
        - 어제꺼는 반영이 안되어있으니 어제 돌린걸로 train하려면 직접 수정 필요
    - 단일 `train.py`에서 hyperparameter yaml 가져올 수 있도록 수정

### 의문
1. pruning과 dropout하면 MACs 가 실제로 줄어드는가?

2. pruning과 dropout 차이
- Pruning은 한번 잘라낸 가지를 다시 복원하지 않지만 Dropout은 weight의 사용을 껏다 켰다를 반복한다.

### 발표
- MobileNetv2 (아름)
- tensor decomposition (종헌)

[dwconv 그림](https://eehoeskrap.tistory.com/431)

[1x1 convolution](https://hwiyong.tistory.com/45)

ResNet의 identity mapping을 이해하고 싶다면 [이 논문](https://arxiv.org/pdf/1603.05027.pdf)을 봐주세요. 🤩

### 로봇
- 4GB 라즈베리파이 활용
- 경량화된 모델과 아닌 모델간의 성능 비교
- 조교님께 여쭤볼 수 있으면 여쭤보기
- 아이디어 생각하기

### 사물인터넷 구성요소 
1. 마이크로컨트롤러 장치 (MCU, Microcontroller Unit) ex) 아두이노 
- CPU의 기능을 하는 핵심 장치와 주변 장치들을 포함하는 칩셉
- 소프트웨어를 다운로드하면 CODING의 목적대로 동작하며, 주변 장치들을 제어함
2. MPU (Microcontroller Unit)
3. 센서 
4. 액츄에이터

### 오늘 일정
- 밤에 다같이 돌리기

### 회고록
Day 6
주말 피드백
코드 방향성 설정

Day 7
optuna 연동 구현
wandb 연동 구현
tensor decomposition 개념 알아보기

Day 8
optuna 버그 수정
DB 연동하는 방법 찾기

Day 9
DB 연동하여 다같이 trial 돌리기
멘토링 사이드프로젝트

Day 10
tensor decomposition 구현

(+)잘했던것, 좋았던것, 계속할것
- 아침회의를 통해 하루를 일찍시작한 것.
- 아침 강의 같이 들은 것
- gather town을 통해 언제든지 편하게 질문할 수 있는 분위기
- 노션/깃헙 등을 통해 협업 프로세스를 구체화한 것
- Special mission/Daily mission을 다같이 하고 서로 공유함
- Notion을 사용하여 공부할 것을 정리함
- 깃헙 PR, review, approve 기능을 적극적으로 사용
- 종헌이의 수학강의
- optuna 사용, wandb 연결, DB 연결, 이론 공부 등에 대해 분업을 한 후 결과물을 공유하고 합치므로써 시간을 절약할 수 있었다.
- config 파일을 사용해서 모델 실험 관리
- DB를 연동하여 모든 조원이 하나의 최적화를 수행하여 빠르게 최적화할 수 있었다.

(-)잘못했던것, 아쉬운것, 부족한것 à 개선방향
- 너무 늦게잤다.
- 테트리스가 너무 재밌다.

(!) 도전할것, 시도할것
- tensor decomposition 적용
- pruning 적용
- 경량화로 엣지 디바이스에 올릴만한거 구상하기
- auto albumentation @ optuna
- rand augmentation

(?) 키워드(공부한것, 알게된것, 느낀점)
- wandb
- optuna
- postgresql
- Tensor decomposition
